### PR TITLE
feat(users): handle email url for users in different tenancies

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Add support for user in tenants configs.
- Handle email URLs for different tenancies

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Closes [#6808](https://github.com/juspay/hyperswitch/issues/6808)

## How did you test it?
Tested locally with email feature flag on, for different tenancies the emails we were getting, we were landing onto to the URL specified in the tenant's config.
We specified the following in the local configs.
```
[multitenancy]
enabled = true
global_tenant = { schema = "public", redis_key_prefix = "", clickhouse_database = "default"}

[multitenancy.tenants.public]
base_url = "http://localhost:8080"
schema = "public"
redis_key_prefix = ""
clickhouse_database = "default"

[multitenancy.tenants.public.user]
control_center_url =  "http://localhost:9000"

[multitenancy.tenants.test]
base_url = "http://localhost:8080"
schema = "public"
redis_key_prefix = ""
clickhouse_database = "default"

[multitenancy.tenants.test.user]
control_center_url =  "http://localhost:9001"
```
We were getting the mails
image

For public Tenant: (**http://localhost:9000**/user/verify_email?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6ImFwb29ydmRpeGl0ODhAZ21haWwuY29tIiwiZmxvdyI6InZlcmlmeV9lbWFpbCIsImV4cCI6MTczNDAwNDg0MSwiZW50aXR5IjpudWxsfQ.j0RlE85yahAALsMccb2uQtt34JIrFV31t3eNYfO5Sdc)

For test Tenant: (**http://locatlhost:9001**/user/verify_email?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6ImFwb29ydmRpeGl0ODgrMUBnbWFpbC5jb20iLCJmbG93IjoidmVyaWZ5X2VtYWlsIiwiZXhwIjoxNzM0MDA0OTY4LCJlbnRpdHkiOm51bGx9.KIiSssKT1J0awgOt9C64tx6lSNrc0CbfZPPXpyRXP_8)

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by ThisIsMani — DISMISSED** [review]

**Review by ThisIsMani — APPROVED** [review]

File: crates/router/src/configs/settings.rs [hunk 1]
```diff
@@ -169,6 +169,12 @@ pub struct Tenant {
     pub schema: String,
     pub redis_key_prefix: String,
     pub clickhouse_database: String,
+    pub user: User,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct User {
```
Comment (SanchithHegde) [inline crates/router/src/configs/settings.rs:176]: Can we name the struct to be `TenantUserSettings` or something similar?

File: crates/drainer/src/settings.rs [hunk 2]
```diff
@@ -179,6 +181,12 @@ pub struct Tenant {
     pub schema: String,
     pub redis_key_prefix: String,
     pub clickhouse_database: String,
+    pub user: User,
```
Comment (SanchithHegde) [inline crates/drainer/src/settings.rs:184]: Why should the drainer include this config?

**Review by SanchithHegde — DISMISSED** [review]

**Review by jarnura — APPROVED** [review]

File: crates/router/src/services/email/types.rs [hunk 3]
```diff
@@ -237,7 +245,7 @@ pub struct VerifyEmail {
 /// Currently only HTML is supported
 #[async_trait::async_trait]
 impl EmailData for VerifyEmail {
-    async fn get_email_data(&self) -> CustomResult<EmailContents, EmailError> {
+    async fn get_email_data(&self, base_url: &str) -> CustomResult<EmailContents, EmailError> {
```
Comment (ThisIsMani) [inline crates/router/src/services/email/types.rs:248]: Can we take base URL in the `VerifyEmail` type instead?

**Review by racnan — APPROVED** [review]
put `get_base_url`'s functionality in types or abstract it away.

File: crates/router/src/services/email/types.rs (hunk 3 above)
Comment (apoorvdixit88) [reply crates/router/src/services/email/types.rs:248]: We can take in types. Can be refactored in separate PR.

**Review by ThisIsMani — APPROVED** [review]

**Pull request merged**